src/game_player.cpp

```diff
@@ -613,10 +613,12 @@ bool Game_Player::Move(int dir) {
 		if (!terrain->on_damage_se || (terrain->on_damage_se && (terrain->damage > 0))) {
 			Main_Data::game_system->SePlay(terrain->footstep);
 		}
-		if (terrain->damage > 0) {
+		if (terrain->damage != 0) {
 			for (auto hero : Main_Data::game_party->GetActors()) {
-				if (!hero->PreventsTerrainDamage()) {
-					red_flash = true;
+				if (terrain->damage < 0 || !hero->PreventsTerrainDamage()) {
```

Comment:
did you verify that:

- "Preventing damage" does not prevent heal? (we are talking about RPG Maker 2003 here!)
- "Negative damage" does not flash the screen in a red color?